In [0]:
import numpy as np
import pandas as pd
import seaborn as sns
import statistics as st
from scipy import interp
import matplotlib.pyplot as plt
from sklearn import svm, datasets
from sklearn.model_selection import KFold
from sklearn.metrics import roc_curve, auc
from sklearn import metrics, model_selection
from sklearn.metrics import mean_squared_error
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import StratifiedKFold, GridSearchCV

from sklearn.model_selection import train_test_split
from sklearn.metrics import roc_curve, auc, confusion_matrix, precision_score, recall_score
from sklearn.ensemble import RandomForestClassifier, GradientBoostingClassifier, AdaBoostClassifier

%matplotlib inline

In [2]:
churn=pd.read_csv("https://raw.githubusercontent.com/firmansyah2308/homework_basic_programming/master/churn.csv", encoding='latin1')
churn.tail(10)

state  account_length  ...  number_customer_service_calls  class
4990     28             140  ...                              1      1
4991      3              97  ...                              1      1
4992     26              83  ...                              0      0
4993     49              73  ...                              3      0
4994     27              75  ...                              1      0
4995     11              50  ...                              2      0
4996     49             152  ...                              3      1
4997      7              61  ...                              1      0
4998      7             109  ...                              0      0
4999     46              86  ...                              0      0

[10 rows x 21 columns]

## Q1. Manakah area_code yang paling di data all populasi tsb?

In [12]:
st.mode(churn['area_code'])

415

## Q2. Bagaimanakah karakteristik Variables All Populasi?

In [7]:
churn.describe().class

state  account_length  ...  number_customer_service_calls        class
count  5000.00000      5000.00000  ...                    5000.000000  5000.000000
mean     25.99840       100.25860  ...                       1.570400     0.141400
std      14.80348        39.69456  ...                       1.306363     0.348469
min       0.00000         1.00000  ...                       0.000000     0.000000
25%      13.00000        73.00000  ...                       1.000000     0.000000
50%      26.00000       100.00000  ...                       1.000000     0.000000
75%      39.00000       127.00000  ...                       2.000000     0.000000
max      50.00000       243.00000  ...                       9.000000     1.000000

[8 rows x 21 columns]

- Rata-rata panggilan number_customer_service_calls: 1.57
- Rata-rata LOS account_length customer: 100.3
- Rata-rata day_minutes: 180.3

## Q3. Ada berapa kah jumlah pelanggan yg stay dan churn? dan jumlah rationya?

In [41]:
stayed = churn.groupby('class').phone_number.count()[0]
churned = churn.groupby('class').phone_number.count()[1]
churn_ratio = churned/(stayed+churned)*100

print('Jumlah pelanggan yg stay :', stayed)
print('Jumlah pelanggan yg churn :', churned)
print('Ratio churn :', churn_ratio, "%")

Jumlah pelanggan yg stay : 4293
Jumlah pelanggan yg churn : 707
Ratio churn : 14.14 %


## Q4. Bagaimanakah karakteristik Variables dari pelanggan yang Churn?

In [48]:
churn[churn['class']== 1].describe()

state  account_length  ...  number_customer_service_calls  class
count  707.000000      707.000000  ...                     707.000000  707.0
mean    26.251768      102.332390  ...                       2.254597    1.0
std     14.228766       39.396993  ...                       1.815956    0.0
min      0.000000        1.000000  ...                       0.000000    1.0
25%     15.000000       76.000000  ...                       1.000000    1.0
50%     26.000000      102.000000  ...                       2.000000    1.0
75%     38.000000      126.500000  ...                       4.000000    1.0
max     50.000000      225.000000  ...                       9.000000    1.0

[8 rows x 21 columns]

Pelanggan yang churn memiliki karakteristik sbg berikut:
* Sebagian besar terjadi di area_code : 415
* Rata-rata di LOS 102.3
* Rata-rata day_minutes: 207.9

## Q5. Berapa jumlah pelanggan yg churn di tiap area_code-nya?

In [56]:
churn[churn['class']== 1].groupby('area_code').count()

state  account_length  ...  number_customer_service_calls  class
area_code                         ...                                      
408          177             177  ...                            177    177
415          346             346  ...                            346    346
510          184             184  ...                            184    184

[3 rows x 20 columns]

* Untuk area_code 408: 177 subs
* Untuk area_code 415: 346 subs
* Untuk area_code 501: 184 subs